python_scripts/parameter_tuning_sol_03.py

```diff
@@ -108,80 +103,84 @@
-#     n_iter=200, error_score=np.nan, n_jobs=-1)
-# _ = model_random_search.fit(df_train, target_train)
-# cv_results =  pd.DataFrame(model_random_search.cv_results_)
-# cv_results.to_csv("../figures/randomized_search_results_logistic_regression.csv")
+# We could use `cv_results = model_random_search.cv_results_` to make a
+# parallel coordinate plot as we did in the previous notebook (you are more
+# than welcome to try!). Instead we are going to load the results obtained from
+# a similar search with many more iterations (1,000 instead of 20).
 
 # %%
 cv_results = pd.read_csv(
-    "../figures/randomized_search_results_logistic_regression.csv",
-    index_col=0)
+    "../figures/randomized_search_results_logistic_regression.csv")
+
+# %% [markdown]
+# To simplify the axis of the plot, we will rename the column of the dataframe
+# and only select the mean test score and the value of the hyperparameters.
 
 # %%
-column_results = [f"param_{name}"for name in param_distributions.keys()]
-column_results += ["mean_test_score", "std_test_score", "rank_test_score"]
-
-cv_results = cv_results[column_results].sort_values(
-    "mean_test_score", ascending=False)
-cv_results = (
-    cv_results
-    .rename(columns={
-        "param_logisticregression__C": "C",
-         "param_logisticregression__solver": "solver",
-         "param_logisticregression__penalty": "penalty",
-         "param_columntransformer__cat-preprocessor__drop": "drop",
-         "mean_test_score": "mean test accuracy",
-         "rank_test_score": "ranking"})
-    .astype(dtype={'C': 'float64'})
-)
-cv_results['log C'] = np.log(cv_results['C'])
+column_name_mapping = {
+    "param_logisticregression__C": "C",
+    "param_columntransformer__num-preprocessor__with_mean": "centering",
+    "param_columntransformer__num-preprocessor__with_std": "scaling",
+    "mean_test_score": "mean test accuracy",
+}
+
+cv_results = cv_results.rename(columns=column_name_mapping)
+cv_results = cv_results[column_name_mapping.values()].sort_values(
+    "mean test accuracy", ascending=False)
+
+# %% [markdown]
+# In addition, the parallel coordinate plot from `plotly` expect all data to be
+# numeric. Thus, we convert the boolean indicator informing whether or not the
+# data were centered or scaled into an integer, where True is mapped to 1 and
+# False is mapped to 0.
+#
+# We also take the logarithm of the `C` values to span the data on a broader
+# range for a better visualization.
 
 # %%
-cv_results["drop"] = cv_results["drop"].fillna("None")
-cv_results = cv_results.dropna("index").drop(columns=["solver"])
-encoding = {}
-for col in cv_results:
-    if cv_results[col].dtype.kind == 'O':
-        labels, uniques = pd.factorize(cv_results[col])
-        cv_results[col] = labels
-        encoding[col] = uniques
-encoding
+column_scaler = ["centering", "scaling"]
+cv_results[column_scaler] = cv_results[column_scaler].astype(np.int64)
+cv_results['log C'] = np.log(cv_results['C'])
```

Comment:
To make it easier to reason about C:

```suggestion
cv_results['log C'] = np.log10(cv_results['C'])
```